File: BSvsZP-Common/MessagesTester/MessageEncodingSamples.cs
using System;
using System.Text;
using System.IO;
using Microsoft.VisualStudio.TestTools.UnitTesting;

using Messages;
using Common;

namespace MessagesTester
{
    [TestClass]
    public class MessageEncodingSamples
    {
        [TestMethod]
        public void CreateEncodingSamples()
        {
            StreamWriter writer = new StreamWriter("MessageSamples.txt");

            MessageNumber msgNumber = MessageNumber.Create(100, 120);
            MessageNumber conversationNumber = MessageNumber.Create(200, 240);
            AgentInfo agentInfo = new AgentInfo(10, AgentInfo.PossibleAgentType.BrilliantStudent, new Common.EndPoint("129.123.5.10:1234"))
                                        {
                                            AgentStatus = AgentInfo.PossibleAgentStatus.InGame,
                                            ANumber = "A00001",
                                            FirstName = "Joe",
                                            LastName = "Jones",
                                            Location = new FieldLocation(10, 20),
                                            Points = 100,
                                            Strength = 200,
                                            Speed = 1.2
                                        };


            AckNak ackNak = new AckNak(Reply.PossibleStatus.Success, agentInfo, "Test Message")
                                        {
                                            MessageNr = msgNumber,
                                            ConversationId = conversationNumber,
                                            IntResult = 99,
                                            Note = "Test Note"
                                        };
            writer.WriteLine("AckNak");
            writer.WriteLine("\tMessageNr={0}", ackNak.MessageNr.ToString());
            writer.WriteLine("\tConversationId={0}", ackNak.ConversationId.ToString());
            writer.WriteLine("\tReplyType={0}", ackNak.ReplyType);
            writer.WriteLine("\tAckNak Status={0}", ackNak.Status);
            writer.WriteLine("\tAgent Info:");
            writer.WriteLine("\t\tId={0}", agentInfo.Id);
            writer.WriteLine("\t\tAgentStatus={0}", agentInfo.AgentStatus);
            writer.WriteLine("\t\tANumber={0}", agentInfo.ANumber);
            writer.WriteLine("\t\tFirstName={0}", agentInfo.FirstName);
            writer.WriteLine("\t\tLastName={0}", agentInfo.LastName);
            writer.WriteLine("\t\tLocation={0}", agentInfo.Location.ToString());
            writer.WriteLine("\t\tPoints={0}", agentInfo.Points);
            writer.WriteLine("\t\tStrength={0}", agentInfo.Strength);
            writer.WriteLine("\t\tSpeed={0}", agentInfo.Speed);

            ByteList byteList = new ByteList();
            ackNak.Encode(byteList);

            writer.WriteLine("");
            writer.WriteLine("Encoding:");
            writer.WriteLine(byteList.CreateLogString());
            writer.WriteLine("");
            writer.WriteLine("------------------------------------");
            writer.WriteLine("");

            JoinGame joinGame = new JoinGame(20, agentInfo)
                                        {
                                            MessageNr = msgNumber,
                                            ConversationId = conversationNumber,
                                        };

            writer.WriteLine("JoinGame");
            writer.WriteLine("\tMessageNr={0}", joinGame.MessageNr.ToString());
            writer.WriteLine("\tConversationId={0}", joinGame.ConversationId.ToString());
            writer.WriteLine("\tGameId={0}", joinGame.GameId);
            writer.WriteLine("\tAgent Info:");
            writer.WriteLine("\t\tId={0}", agentInfo.Id);
            writer.WriteLine("\t\tAgentStatus={0}", agentInfo.AgentStatus);
            writer.WriteLine("\t\tANumber={0}", agentInfo.ANumber);
            writer.WriteLine("\t\tFirstName={0}", agentInfo.FirstName);
            writer.WriteLine("\t\tLastName={0}", agentInfo.LastName);
            writer.WriteLine("\t\tLocation={0}", agentInfo.Location.ToString());
            writer.WriteLine("\t\tPoints={0}", agentInfo.Points);
            writer.WriteLine("\t\tStrength={0}", agentInfo.Strength);
            writer.WriteLine("\t\tSpeed={0}", agentInfo.Speed);

            byteList = new ByteList();
            joinGame.Encode(byteList);

            writer.WriteLine("");
            writer.WriteLine("Encoding:");
            writer.WriteLine(byteList.CreateLogString());
            writer.WriteLine("");
            writer.WriteLine("------------------------------------");
            writer.WriteLine("");

            // TODO: All of the other message types
        }
    }
}

File: BSvsZP-Common -Java/MessagesTester/MessageEncodingSamples.java
package MessagesTester;

import java.io.BufferedWriter;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.OutputStreamWriter;
import java.io.Writer;
import org.junit.Test;
import org.omg.CORBA.portable.ApplicationException;
import Common.AgentInfo;
import Common.ByteList;
import Common.FieldLocation;
import Common.MessageNumber;
import Messages.AckNak;
import Messages.JoinGame;
import Messages.Reply;

public class MessageEncodingSamples {

	@Test
	public void CreateEncodingSamples() throws Exception {
		Writer writer = null;
		writer = new BufferedWriter(new OutputStreamWriter(new FileOutputStream("MessageSamples.txt"), "UTF-8"));
		
		MessageNumber msgNumber = MessageNumber.Create((short)100, (short)120);
        MessageNumber conversationNumber = MessageNumber.Create((short)200, (short)240);
		
        AgentInfo agentInfo = new AgentInfo((short)10, AgentInfo.PossibleAgentType.BrilliantStudent, new Common.EndPoint("129.123.5.10:1234"));
        agentInfo.setAgentStatus(AgentInfo.PossibleAgentStatus.InGame);
        agentInfo.setANumber("A00001");
        
        agentInfo.setFirstName("Joe");
        agentInfo.setLastName("Jones");
        agentInfo.setLocation(new FieldLocation((short) 10, (short) 20));
        
        agentInfo.setPoints((Double) 100.0);
        agentInfo.setStrength((Double) 200.0);
        agentInfo.setSpeed(1.2);
     


        AckNak ackNak = new AckNak(Reply.PossibleStatus.Success, agentInfo, "Test Message");
        ackNak.setMessageNr(msgNumber);
        ackNak.setConversationId(conversationNumber);
        ackNak.setIntResult(99);
        ackNak.setNote("Test Note");
        
        writer.write("AckNak");
        writer.write("\n" + "\tMessageNr = " + ackNak.getMessageNr().toString());
        writer.write("\n" + "\tConversationId = " +  ackNak.getConversationId().toString());
        writer.write("\n" + "\tReplyType = " +  ackNak.getReplyType());
        writer.write("\n" + "\tAckNak Status = " +  ackNak.getStatus());
        writer.write("\n" + "\tAgent Info:");
        writer.write("\n" + "\t\tId = " + agentInfo.getId());
        writer.write("\n" + "\t\tAgentStatus = " + agentInfo.getAgentStatus());
        writer.write("\n" + "\t\tANumber = " +  agentInfo.getANumber());
        writer.write("\n" + "\t\tFirstName = " + agentInfo.getFirstName());
        writer.write("\n" + "\t\tLastName = " + agentInfo.getLastName());
        writer.write("\n" + "\t\tLocation (X = " + agentInfo.getLocation().getX() + ", Y = " +  agentInfo.getLocation().getY() + ")");
        
        writer.write("\n" + "\t\tPoints = " + agentInfo.getPoints());
        writer.write("\n" + "\t\tStrength = " + agentInfo.getStrength());
        writer.write("\n" + "\t\tSpeed = " + agentInfo.getSpeed());
        
        ByteList byteList = new ByteList();
        ackNak.Encode(byteList);

        writer.write("");
        writer.write("\n" + "Encoding:");
        printAll(byteList, writer);
        writer.write("\n" );
        writer.write("------------------------------------");
        writer.write("\n");

        JoinGame joinGame = new JoinGame((short)20, agentInfo);
        joinGame.setMessageNr(msgNumber);
        joinGame.setConversationId(conversationNumber);
                                    
        writer.write("JoinGame");
        writer.write("\n" + "\tMessageNr = " +  joinGame.getMessageNr().toString());
        writer.write("\n" + "\tConversationId = " + joinGame.getConversationId().toString());
        writer.write("\n" + "\tGameId = " + joinGame.getGameId());
        writer.write("\n" + "\tAgent Info:");
        writer.write("\n" + "\t\tId = " + agentInfo.getId());
        writer.write("\n" + "\t\tAgentStatus = " + agentInfo.getAgentStatus());
        writer.write("\n" + "\t\tANumber = " + agentInfo.getANumber());
        writer.write("\n" + "\t\tFirstName = " + agentInfo.getFirstName());
        writer.write("\n" + "\t\tLastName = " + agentInfo.getLastName());
        writer.write("\n" + "\t\tLocation = " + agentInfo.getLocation().ToString());
        writer.write("\n" + "\t\tPoints = " + agentInfo.getPoints());
        writer.write("\n" + "\t\tStrength = " + agentInfo.getStrength());
        writer.write("\n" + "\t\tSpeed = " + agentInfo.getSpeed());
	
        
        ByteList byteList2 = new ByteList();
        joinGame.Encode(byteList2);

        writer.write("");
        writer.write("Encoding:");
        printAll(byteList2, writer);
        writer.write("");
        writer.write("------------------------------------");
        writer.write("");
		writer.close();
		
	}

	public static void printAll(ByteList bytes, Writer writer) throws ApplicationException, IOException
	{
		  byte[] b = bytes.GetBytes(bytes.getLength());
	        for(int i =0; i<b.length; ++i)
	        	writer.write( i + ":" + b[i] + " ");
	       
	}
}
